src/main/java/edu/cornell/kfs/pmw/document/validation/impl/PaymentWorksVendorGlobalRule.java

```diff
@@ -0,0 +1,162 @@
+                    PaymentWorksVendorGlobalAction.RESTAGE_FOR_UPLOAD.actionLabel);
+            return false;
+        }
+        return true;
+    }
+
+    private boolean checkPaymentWorksVendorChanges() {
+        final List<PaymentWorksVendorGlobalDetail> vendorDetails = paymentWorksVendorGlobal.getVendorDetails();
+        if (CollectionUtils.isEmpty(vendorDetails)) {
+            putFieldError(PaymentWorksPropertiesConstants.VENDOR_DETAILS,
+                    PaymentWorksKeyConstants.ERROR_PAYMENTWORKS_VENDOR_GLOBAL_DETAILS_EMPTY);
+            return false;
+        }
+
+        boolean success = true;
+        int i = -1;
+        final Set<Integer> encounteredPmwVendors = new HashSet<>();
+
+        for (final PaymentWorksVendorGlobalDetail vendorDetail : vendorDetails) {
+            i++;
+            final String errorPathPrefix = buildVendorDetailsErrorPathPrefix(i);
+            final String pmwVendorIdErrorPath = errorPathPrefix + PaymentWorksPropertiesConstants.PMW_VENDOR_ID;
+            if (!encounteredPmwVendors.add(vendorDetail.getPmwVendorId())) {
+                putFieldError(pmwVendorIdErrorPath,
+                        PaymentWorksKeyConstants.ERROR_PAYMENTWORKS_VENDOR_GLOBAL_DETAILS_DUPLICATE,
+                        String.valueOf(vendorDetail.getPmwVendorId()));
+                success = false;
+            }
+
+            final PaymentWorksVendor pmwVendor = vendorDetail.getPmwVendor();
+            if (ObjectUtils.isNull(pmwVendor)) {
+                putFieldError(pmwVendorIdErrorPath,
+                        PaymentWorksKeyConstants.ERROR_PAYMENTWORKS_VENDOR_GLOBAL_DETAILS_NOT_FOUND,
+                        String.valueOf(vendorDetail.getPmwVendorId()));
+                success = false;
+            } else if (paymentWorksVendorGlobal.getActionType() == PaymentWorksVendorGlobalAction.RESTAGE_FOR_UPLOAD) {
+                success &= checkPaymentWorksVendorChangeForRestaging(pmwVendor, errorPathPrefix);
+            }
+        }
+
+        return success;
+    }
+
+    private boolean checkPaymentWorksVendorChangeForRestaging(
+            final PaymentWorksVendor pmwVendor, final String errorPathPrefix) {
+        final String uploadStatusErrorPath = errorPathPrefix
+                + PaymentWorksPropertiesConstants.PaymentWorksVendor.SUPPLIER_UPLOAD_STATUS;
+
+        switch (StringUtils.defaultString(pmwVendor.getSupplierUploadStatus())) {
+            case SupplierUploadStatus.VENDOR_UPLOADED :
+            case SupplierUploadStatus.UPLOAD_FAILED :
+                return true;
+
+            case SupplierUploadStatus.READY_FOR_UPLOAD :
+                putFieldError(uploadStatusErrorPath,
+                        PaymentWorksKeyConstants.ERROR_PAYMENTWORKS_VENDOR_GLOBAL_DETAILS_UPLOAD_STATUS_MATCH,
+                        new String[] { String.valueOf(pmwVendor.getId()), pmwVendor.getSupplierUploadStatus() });
+                return false;
```

Comment:
Does this case need to be an error?

Reply:
There's no point in restaging a vendor that's already properly staged for upload, so it would be helpful to report that to the user. Do you think this should be downgraded to a warning instead?

Reply:
I agree there is no point in re-staging an item that is already staged.  There is also no harm in that action either.
Maybe a warning is a good compromise.
My thought is user experience, and making this as easy as possible